src/plugin/withSentryAndroid.ts

```diff
@@ -0,0 +1,40 @@
+import { ConfigPlugin, withAppBuildGradle, withDangerousMod } from '@expo/config-plugins';
+import * as path from 'path';
+
+import { writeSentryPropertiesTo } from './withSentryIOS';
+
+export const withSentryAndroid: ConfigPlugin<string> = (config, sentryProperties: string) => {
+  config = withAppBuildGradle(config, (config) => {
+    if (config.modResults.language === 'groovy') {
+      config.modResults.contents = modifyBuildGradle(config.modResults.contents);
+    } else {
+      throw new Error(
+        'Cannot configure Sentry in the app gradle because the build.gradle is not groovy'
+      );
+    }
+    return config;
+  });
+  return withDangerousMod(config, [
+    'android',
+    (config) => {
+      writeSentryPropertiesTo(
+        path.resolve(config.modRequest.projectRoot, 'android'),
+        sentryProperties
+      );
+      return config;
+    },
+  ]);
+};
+
+function modifyBuildGradle(buildGradle: string) {
```

Comment:
```suggestion
function modifyAppBuildGradle(buildGradle: string) {
```
makes it easier to follow along. Maybe also add a quick doc-block that says "writes to the app/build.gradle" for ppl who are trying to quickly understand the setup.